### ISSUE TITLE
Fixed Navigation Header: cleanup and docs

### ISSUE THREAD

**Issue by cpapazoglou** [Automattic/wp-calypso#62166]
After looking at https://github.com/Automattic/wp-calypso/pull/62080/files, I found at about this code 
https://github.com/Automattic/wp-calypso/pull/58179/files/ada2ecd94d2df7601a63ea07b1434d51e70631a4#diff-d717431d31dbd940b8fb904c76f0d160e964e737d0f90780328884bcb38971d8R67-R96

Can we add some comments about what this `useEffect` does for posterity? Also, I noticed a new prop which should be added in the `readme` https://github.com/Automattic/wp-calypso/blob/831b4c593e979f6b214724ff4693c2aebbc9fa26/client/components/fixed-navigation-header/README.md#L25-L29 

Moreover, it seems to me that it could have been the parent component responsibility to render or not the `actionElement` (perhaps by using a similar `useEffect` rather than having to pass the `contentRef`. Is there a reason for choosing to place this functionality here?

_Originally posted by @cpapazoglou in https://github.com/Automattic/wp-calypso/pull/58179#discussion_r831852420_

**Comment by allilevine** [comment]
> Can we add some comments about what this useEffect does for posterity?

Sounds good! For context, we needed to add a button that appears on scroll. It's in use on /woocommerce-installation

Acceptance criteria:
- [ ] Code comment about what the `useEffect` is used for.
- [ ] The `contentRef` prop is documented in the readme.
- [ ] We've investigated whether this functionality can be placed in the parent component or removed entirely (the landing page is a lot shorter now.)

**Comment by Rolilink** [comment]
My gut tell me that we don't actually need the `useEffect` logic and that we can do this through the parent component, I am also migrating this logic into an `useSticky` hook  on #62532.

**Comment by lsl** [comment]
> My gut tell me that we don't actually need the `useEffect` logic and that we can do this through the parent component

Moved this to the parent https://github.com/Automattic/wp-calypso/pull/62548 still needed access to the header component but `forwardRef` was able to clean that up a bit. 

I didn't drop the useEffect, I think we need it to handle unregistering the scroll handler?

- [x] ~Code comment about what the `useEffect` is used for.~ - moved to parent
- [x] ~The `contentRef` prop is documented in the readme.~ - removed `contentRef`, still had to expose the header using `forwardRef` but documenting that didn't make sense to me, feel free to suggest something and I'll add it.
- [x] We've investigated whether this functionality can be placed in the parent component or removed entirely (the landing page is a lot shorter now.) - moved to parent, still need for the mobile view imo.
